src/Compilers/CSharp/Portable/Symbols/MetadataOrSourceAssemblySymbol.cs

```diff
@@ -21,12 +21,44 @@ internal abstract class MetadataOrSourceAssemblySymbol
         /// </summary>
         /// <remarks></remarks>
         private NamedTypeSymbol[] _lazySpecialTypes;
+        private ThreeState _lazyRuntimeSupportsDefaultInterfaceImplementation = ThreeState.Unknown;
 
         /// <summary>
         /// How many Cor types have we cached so far.
         /// </summary>
         private int _cachedSpecialTypes;
 
+        internal override bool RuntimeSupportsDefaultInterfaceImplementation
+        {
+            get
+            {
+                if ((object)CorLibrary == this)
+                {
+                    if (!_lazyRuntimeSupportsDefaultInterfaceImplementation.HasValue())
+                    {
+                        // PROTOTYPE(DefaultInterfaceImplementation): Implement real detection if the feature is supported by the runtime
+                        // For now we assume that it is supported by default
+                        _lazyRuntimeSupportsDefaultInterfaceImplementation = ThreeState.True;
```

Comment:
"Immediately visible" is an inconvenient term. Memory writes _are_ immediately visible, but that is not enough in many cases because of read/write ordering.

The important part here is consensus - all readers read the same value. We have that in this pattern.

Yes, per our memory model, write/read of an int is atomic. And since we deal with a single value assigned the same value, the ordering is irrelevant.